t/plugin/opa2.t

```diff
@@ -0,0 +1,206 @@
+        $block->set_value("no_error_log", "[error]");
+    }
+
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+});
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: setup all-in-one test
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/upstreams/u1',
+                ngx.HTTP_PUT,
+                [[{
+                    "nodes": {
+                        "127.0.0.1:1980": 1
+                    },
+                    "type": "roundrobin"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+                return
+            end
+            ngx.say(body)
+
+            local code, body = t('/apisix/admin/consumers',
+                ngx.HTTP_PUT,
+                [[{
+                    "username": "test",
+                    "plugins": {
+                        "key-auth": {
+                            "disable": false,
+                            "key": "test-key"
+                        }
+                    }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+                return
+            end
+            ngx.say(body)
+
+            local code, body = t('/apisix/admin/services/s1',
+                ngx.HTTP_PUT,
+                [[{
+                    "name": "s1",
+                    "plugins": {
+                        "key-auth": {
+                            "disable": false
+                        }
+                    }
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+                return
+            end
+            ngx.say(body)
+
+            local code, body = t('/apisix/admin/routes/1',
+                ngx.HTTP_PUT,
+                [[{
+                    "plugins": {
+                        "opa": {
+                            "host": "http://127.0.0.1:8181",
+                            "policy": "debug",
+                            "with_route": true,
+                            "with_consumer": true,
+                            "with_service": true
+                        }
+                    },
+                    "upstream_id": "u1",
+                    "service_id": "s1",
+                    "uri": "/hello"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+                return
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body eval
+"passed\n" x 4
+
+
+
+=== TEST 2: hit route (test route data)
+--- request
+GET /hello
+--- more_headers
+test-header: only-for-test
+apikey: test-key
+--- error_code: 403
+--- response_body eval
+qr/\"route\":/ and qr/\"id\":\"r1\"/ and qr/\"plugins\":\{\"opa\"/ and
```

Comment:
`and` is ok?

first time I've seen it written this way🤣

Reply:
I tested it and it should work. The first time I tried this strange usage, and I have never used it. 😂😂